### PR TITLE
Hilla: update the boolean binding example for `<vaadin-select>`

### LINKED ISSUE #1474: Update the boolean binding example for <vaadin-select>
Now that the Hilla form binder supports binding boolean fields to `<vaadin-select>`, we should include it in the example here: https://hilla.dev/docs/data-binding/vaadin-components/#binding-data-to-boolean-fields

The source of the article: https://github.com/vaadin/docs/blob/hilla/articles/data-binding/vaadin-components.asciidoc

See the feature ticket for the usage code example: https://github.com/vaadin/hilla/issues/177

Use the `selectRenderer` included in `23.1` `<vaadin-select>`: https://github.com/vaadin/web-components/pull/3775

### PR DESCRIPTION
**Pull request by cromoteca**
Add a code snippet to demonstrate how to bind a boolean to a `select`.

Closes #1474

### CODE REVIEW

**Comment by cromoteca** [comment]
The added example won't run in an application: see vaadin/web-components#4059

**Comment by cromoteca** [comment]
The new code snippet needs vaadin/web-components#4064 to work

File: articles/data-binding/vaadin-components.asciidoc [hunk 1]
```diff
@@ -102,8 +102,9 @@ render() {
 
 === Binding Data to Boolean Fields
 
-There are two elements in Hilla to deal with booleans: `vaadin-checkbox` and `vaadin-radio-button`.
-Both work fine with form binders, but neither of them has validation and failure styling.
+There are three elements in Hilla to deal with booleans: `vaadin-checkbox`, `vaadin-radio-button`, and `vaadin-select`.
+
+The former two work fine with form binders, but neither of them has validation and failure styling.
```
Comment (dougjewsbury) [inline articles/data-binding/vaadin-components.asciidoc:107]:
```suggestion
The first two of these work fine with form binders, but neither of them has validation and failure styling.
```

File: articles/data-binding/vaadin-components.asciidoc [hunk 2]
```diff
@@ -127,6 +128,23 @@ render() {
 }
 ----
 
+The latter can be bound to a boolean as in the following snippet:
```
Comment (dougjewsbury) [inline articles/data-binding/vaadin-components.asciidoc:131]:
```suggestion
`vaadin-select` can be bound to a boolean as in the following snippet:
```

**Review by dougjewsbury — COMMENTED** [review]
AsciiDoc looks good.

**Review by platosha — APPROVED** [review]

**Pull request merged**